### DISCUSSION TITLE
Shutter/Blinder with Sonoff 4CH Pro and Somfy Sender

### DISCUSSION THREAD

**Discussion by mor3dr3ad** [arendst/Tasmota#20864] [Q&A] [answered]
Hello,

I have a question that I cannot solve on my own: I have a somfy awning that uses a proprietary protocol (IO) and thus cannot be easily integrated into my Home Assistant instance. 

However, I have found a bypass using a Somfy in-wall transmitter that can communicate with the awning motor: https://shop.somfy.de/unterputz-sender-io.html. The sender module has two buttons (S1 and S2), one for up, one for down.

I have hooked this up to a Sonoff 4CH Pro flashed with tasmota (the sender is potential free), S1 is hooked to relay 1 and S2 to relay 2.  Going up and going down works well .

However, I have two problems here: 

1) to stop the awning at a certain point, both buttons need to be pressed. I do not know how I can implement this via the shutter functionality of Tasmota. Is there an option to have a 'virtual' third button by combining the two relays? 

2( obviously, there is no load on the sender module and thus I do not know how I would calibrate the shutter and get back the position of the awning. 

Has anyone implemented something similar and can give me a hint? 

Edited for diagram:

    ------------------   Relay 1 --------------  S1 ----------                                     ------------------------
    |    Sonoff   |                \  _________ COM  | SOMFY   |   ----------- radio ---------    |        Awning      |
    |      4CH     |               /                 | Transm. |                                             -----------------------
    ------------------  Relay 2 ---------------- S2  ---------
        /  /  
    --L /
    ---N

**Comment by Jason2866** [comment]
ping @stefanbode

**Reply by stefanbode** [reply]
We recently had an enhancement for a third relay to stop the whole thing. Do you need to hold the button or is a small impulse enough? In this case you can try the new function to get a stop pulse on relay 3. if you need to hold the relay this is a different beast. 

Check for the pulsetime option on all three relays and checkmate recent documentation. I might remember the new option is called shutterstoprelay1 1

But please check documentation

**Comment by stefanbode** [comment]
Just an idea, but maybe next time you start with the Shelly device directly. Should be same price range but can be flashed with tasmota directly.

**Reply by mor3dr3ad** [reply]
Hey thank you. But what Shelly device are you referring to? There is no shelly device involved.

I am using a Sonoff 4CH, because it is the only device with dry-contact capabilities I had at home. For Shelly, I would need 2 devices...

**Reply by Noschvie** [reply]
There is no difference flashing Tasmota to "Shelly device" and Sonoff 4CH Pro, isn't it?

**Reply by sfromis** [reply]
The flashing process is basically the same for ESP chips, as long as you match the specific ESP8266 or ESP32 series. Of course, different hardware may have different ways of setting up the wiring, and many Shelly devices also have the possibility for OTA flashing without using a serial connection, and/or have an externally accessible port for flashing without cracking it open.

**Comment by Noschvie** [comment, marked as answer]
How have you configured the "Somfy in-wall transmitter", should be Mode 3 (Rollladen) or Mode 4 (Jalousie)
No need to press both buttons.
Isn't it?

![grafik](https://github.com/arendst/Tasmota/assets/48217146/5119ca8d-405a-4ebe-8f4e-6dbaef90929f)

Link https://downloads.somfy.de/frontend/getcatalog.do?catalogId=175046&catalogVersion=1&lang=de#page_2

**Reply by mor3dr3ad** [reply]
Thank you, this was actually the right hint. I had configured Mode 4 (Jalousie), turns out this was the wrong mode for me as I only have a simple awning. However, if one had a venetian blind (Jalousie), this problem would persist.

**Reply by stefanbode** [reply]
If you use mode 3 then I do it see any reason for a relay 3. it should be sufficient on the sonoff 4ch to define a shutter with shuttermode 1. so the most easiest. 

From the image you showed there is no need to press both buttons simultaneously. It says move up Down until release. This is exactly what shuttermode 1 is doing.

**Comment by mor3dr3ad** [comment]
Thank you, @Noschvie. This was the right hint for me. However, this would still persist with the Mode 4 on the Somfy sender unit.

**Reply by Noschvie** [reply]
For mode 4 (Jalousie) maybe you should detach the buttons from the relais, config the inputs for short and long pressed and use rules to control the outputs with short and long pulsetime settings. Sorry, but I'm not an expert for those details.
